### PR TITLE
Make Scylla run in production mode

### PR DESCRIPTION
**Pull request by christos-h**
## Motivation

Scylla is running in developer mode causing performance issues and crashes for docker-compose deployment.

## Proposal

Switch developer mode off. Also enable the `--overprovisioned` flag so that Scylla doesn't hog all the host machine's resources.

## Test Plan

Tested manually. Logs show dev mode is off.

## Release Plan

- These changes should be backported to the latest `testnet` branch, then
    - be released in a validator hotfix.

### CODE REVIEW

**Comment by ma2bd** [comment]
@christos-h @MathieuDutSik Does it impact Scylladb tests? If yes, can we patch the regexp in CI so that the job triggered on this PR (and future ones touching docker) ?

**Comment by ma2bd** [comment]
> @christos-h @MathieuDutSik Does it impact Scylladb tests? If yes, can we patch the regexp in CI so that the job is triggered on this PR (and future ones touching docker) ?

Ok maybe not, actually

**Comment by ma2bd** [comment]
nit: Reading the title of the PR, we don't know if the statement is true before or after the changes

**Review by ma2bd — APPROVED** [review]

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -47,6 +47,8 @@ jobs:
         uses: arduino/setup-protoc@v1
         with:
           repo-token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Update aio-max-nr
+        run: echo 1048576 > /proc/sys/fs/aio-max-nr
```
Comment (ma2bd) [inline .github/workflows/docker.yml:51]: Nice!

Comment (christos-h) [reply .github/workflows/docker.yml:51]: Yeah - this doesn't work on GH runners unfortunately.

**Comment by christos-h** [comment]
Closing in favor of #2702.

**Pull request closed without merging**